### PR TITLE
Relax the `gast` version requirement

### PR DESCRIPTION
**Pull request by thomas-bouvier**
## Category: Other

I'm trying to fix [this issue](https://github.com/NVIDIA/DALI/issues/4865). A first attempt has been submitted [here](https://github.com/NVIDIA/DALI/pull/4869) by @klecki.

Careful, this PR has not been tested yet.

## Description:

Remove the limit on `gast` version.
Handle the missing `type_comment` parameter in `gast.Assign` usages with wrapper `gast_util.compat_assign`.

### Tests:

!!! This PR has to be tested !!!
- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [x] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/python/nvidia/dali/_autograph/pyct/gast_util.py [hunk 1]
```diff
@@ -54,6 +55,14 @@ def _is_ellipsis_gast_3(node):
   return isinstance(node, gast.Constant) and node.value == Ellipsis
 
 
+def compat_assign(targets, value, type_comment):
+  """Wraps around gast.Assign to use same function signature across versions."""
+  if GAST2 or GAST3 or GAST4:
```
Comment (JanuszL) [inline dali/python/nvidia/dali/_autograph/pyct/gast_util.py:60]: How about:
```suggestion
  from distutils.version import LooseVersion
  if LooseVersion(gast.__version__) < LooseVersion("0.5")
```

**Comment by JanuszL** [comment]
Hi @thomas-bouvier,

Thank you for your PR. It looks good. I have only a minor suggestion.

**Comment by thomas-bouvier** [comment]
I used `LooseVersion` objects for version checking. I did not even run the code, should I run tests on my end? Thanks

**Comment by klecki** [comment]
> I used `LooseVersion` objects for version checking. I did not even run the code, should I run tests on my end? Thanks

@thomas-bouvier 
I will handle the tests, thanks for the PR!

**Comment by klecki** [comment]
I tested the changes, and with few tweaks I managed to make it working with following versions: `"0.5.4" "0.5.3" "0.5.2" "0.5.0" "0.4.0" "0.3.3"`

Anything older than 0.3.2 is already broken in the tests, so I bumped the required version to `0.3.3`.
Btw, `0.5.1` has the validation for the argument count broken, as it doesn't include anything that is passed as a named keyword argument.

**Comment by klecki** [comment]
!build

**Comment by thomas-bouvier** [comment]
Can I safely rebase the branch?

**Comment by klecki** [comment]
Sure, please do.

**Comment by thomas-bouvier** [comment]
Done :)

**Comment by klecki** [comment]
!build

**Review by klecki — APPROVED** [review]

File: dali/python/nvidia/dali/_autograph/pyct/gast_util.py [hunk 2]
```diff
@@ -16,20 +16,34 @@
...
-def _is_constant_gast_2(node):
-  return isinstance(node, (gast.Num, gast.Str, gast.Bytes, gast.Ellipsis,
-                           gast.NameConstant))
-
-
-def _is_constant_gast_3(node):
+from distutils.version import LooseVersion
+
+
+def get_gast_version():
```
Comment (JanuszL) [inline dali/python/nvidia/dali/_autograph/pyct/gast_util.py:24]: 👍

**Review by JanuszL — APPROVED** [review]

**Pull request merged**